### PR TITLE
use id model in extent substitution

### PR DESCRIPTION
**Pull request by liqiangxl**
While investigating #2702, I found we can also revise  `exactMappedExtentSubstitution` using IdModel which detects more replacements than `ExactLogicalDomainMap`.

**Before this PR**, `exactMappedExtentSubstitution` uses `ExactLogicalDomainMap`:
For a fusion with:
```
Inputs:
  T0_g[ iS0{i0} ], float
  T1_g[ iS1{i2} ], float
Outputs:
  T4_g[ iS10{32}, iS11{( ceilDiv(i0, 32) )} ], float

%kernel_math {
T2_l[ iS4{32}rf, iS5{( ceilDiv(i0, 32) )}rf ] = view( T0_g[ iS0{i0} ] )
T3_l[ iS8{32}rf, iS9{( ceilDiv(i2, 32) )}rf ] = view( T1_g[ iS1{i2} ] )
T4_g[ iS10{32}, iS11{( ceilDiv(i0, 32) )} ]
   = T2_l[ iS4{32}rf, iS5{( ceilDiv(i0, 32) )}rf ]
   + T3_l[ iS8{32}rf, iS9{( ceilDiv(i2, 32) )}rf ];
} // %kernel_math
```
```
ExactLogicalDomainMap:
disjoint sets{
  { iS0{i0}; iS3{i0}rf }
  { iS1{i2}; iS7{i2}rf }
  { iS9{( ceilDiv(i2, 32) )}rf; iS5{( ceilDiv(i0, 32) )}rf; iS11{( ceilDiv(i0, 32) )} }
  { iS8{32}rf; iS4{32}rf; iS10{32} }
}
```

**After this PR**, `exactMappedExtentSubstitution` uses `exact_graph.disjointValSets()`
```
val_sets: disjoint sets{
  { iS10{32}; iS4{32}rf; iS8{32}rf }
  { iS11{( ceilDiv(i0, 32) )}; iS5{( ceilDiv(i0, 32) )}rf; iS9{( ceilDiv(i2, 32) )}rf }
  { iS3{i0}rf; iS0{i0}; iS7{i2}rf; iS1{i2} }
}
```
**The difference is `exact_graph.disjointValSets()` can detect `{i0} <=>{i2} ` while `ExactLogicalDomainMap` can't.**

 fusion is modified to:
```
Inputs:
  T0_g[ iS0{i0} ], float
  T1_g[ iS13{i0} ], float
Outputs:
  T4_g[ iS10{32}, iS11{( ceilDiv(i0, 32) )} ], float

%kernel_math {
T2_l[ iS4{32}rf, iS5{( ceilDiv(i0, 32) )}rf ] = view( T0_g[ iS0{i0} ] )
T3_l[ iS8{32}rf, iS9{( ceilDiv(i0, 32) )}rf ] = view( T1_g[ iS13{i0} ] )
T4_g[ iS10{32}, iS11{( ceilDiv(i0, 32) )} ]
   = T2_l[ iS4{32}rf, iS5{( ceilDiv(i0, 32) )}rf ]
   + T3_l[ iS8{32}rf, iS9{( ceilDiv(i0, 32) )}rf ];
} // %kernel_math
```

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build --diff

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build --diff

File: csrc/preseg_passes/exact_mapped_extent_substitution.cpp [hunk 1]
```diff
@@ -77,17 +84,23 @@ void exactMappedExtentSubstitution(Fusion* fusion) {
 
 void ExactMappedExtentSubstitutionPass::runPass(Fusion* fusion) {
   if (isDebugDumpEnabled(DebugDumpOption::PreSegmenterLogging)) {
-    debug() << "ExactLogicalDomainMap before " << name() << ":" << std::endl;
-    const auto mapped_sets = ExactLogicalDomainMap(fusion).getMappedSets();
-    debug() << mapped_sets.toString() << std::endl;
+    debug() << "DisjointSets before " << name() << ":" << std::endl;
+    IdModel id_model(fusion, false, false, false);
```
Comment (jacobhinkle) [inline csrc/preseg_passes/exact_mapped_extent_substitution.cpp:88]:
```suggestion
    IdModel id_model(fusion, /*build_graphs=*/false);
```
The `allow_self_mapping` and `validate` arguments default to false anyway.

**Review by jacobhinkle — APPROVED** [review]
I think this pass is basically doing exactly what we want to do in #2714, except it is not giving preference to input tensor extents as is done in that PR. We should probably unify the two at some point.

**Comment by liqiangxl** [comment]
Codediff is expected and failed cases are known issues.

**Pull request merged**